src/datasource.js

```diff
@@ -401,15 +416,40 @@ export default class OCIDatasource {
       }],
       range: this.timeSrv.timeRange()
     }).then((items) => {
-      return this.mapToTextValue(items, 'namespaces')
+      return this.mapToTextValue(items, 'namespaces');
+    });
+  }
+
+  async getResourceGroups(target) {
+    const region = target.region === SELECT_PLACEHOLDERS.REGION ? '' : this.getVariableValue(target.region);
+    const compartment = target.compartment === SELECT_PLACEHOLDERS.COMPARTMENT ? '' : this.getVariableValue(target.compartment);
+    const namespace = target.namespace === SELECT_PLACEHOLDERS.NAMESPACE ? '' : this.getVariableValue(target.namespace);
+    if (_.isEmpty(compartment)) {
+      return this.q.when([]);
+    }
+
+    const compartmentId = await this.getCompartmentId(compartment);
+    return this.doRequest({
+      targets: [{
+        environment: this.environment,
+        datasourceId: this.id,
+        tenancyOCID: this.tenancyOCID,
+        queryType: 'resourcegroups',
+        region: _.isEmpty(region) ? this.defaultRegion : region,
+        compartment: compartmentId,
+        namespace: namespace
+      }],
+      range: this.timeSrv.timeRange()
+    }).then((items) => {
+      return this.mapToTextValue(items, 'resourcegroups');
     });
   }
 
   async getDimensions(target) {
     const region = target.region === SELECT_PLACEHOLDERS.REGION ? '' : this.getVariableValue(target.region);
     const compartment = target.compartment === SELECT_PLACEHOLDERS.COMPARTMENT ? '' : this.getVariableValue(target.compartment);
     const namespace = target.namespace === SELECT_PLACEHOLDERS.NAMESPACE ? '' : this.getVariableValue(target.namespace);
-    
+    const resourcegroup = target.resourcegroup === SELECT_PLACEHOLDERS.RESOURCEGROUP ? '' : this.getVariableValue(target.resourcegroup);
```

Comment:
@japatel 
const resourcegroup = target.resourcegroup === SELECT_PLACEHOLDERS.RESOURCEGROUP ? **'NoResourceGroup'** : this.getVariableValue(target.resourcegroup);
I think NoResourceGroup we can extract to const

Reply:
@VadimKomar In the the new commit I have reverted the ResourceGroup placeholder to 'select resourcegroup'. With this new change we are able to see graphs for both namespaces with or without ResoruceGroup.

Reply:
But we will still need const resourcegroup = target.resourcegroup === SELECT_PLACEHOLDERS.RESOURCEGROUP ? 'NoResourceGroup' : this.getVariableValue(target.resourcegroup);